src/main/java/org/reactome/qa/doisuggester/ReactionlikeEvent.java

```diff
@@ -0,0 +1,349 @@
+
+		try
+		{
+			// For an InstanceEdit to be considered non-Reactome, there must exist at least one author whose project is
+			// not "Reactome"
+			for (GKInstance instanceEdit : instanceEdits)
+			{
+				List<GKInstance> authors = instanceEdit.getAttributeValuesList(ReactomeJavaConstants.author);
+				if (authors != null && !authors.isEmpty())
+				{
+					int i = 0;
+					boolean isNonReactome = false;
+					while (!isNonReactome && i < authors.size())
+					{
+						GKInstance author = authors.get(i);
+						// TODO: Notify Guanming that "project" is not in ReactomeJavaConstants.
+						String project = (String) author.getAttributeValue("project");
+						if (project == null || project.trim().equals("") || !project.trim().equalsIgnoreCase("reactome") )
+						{
+							isNonReactome = true;
+							nonReactomeInstanceEdits.add(instanceEdit);
+						}
+						i++;
+					}
+				}
+//				else
+//				{
+//					// Log a warning about an InstanceEdit with NO authors - there's probably some data cleanup that needs to happen.
+//				}
+			}
+		}
+		catch (InvalidAttributeException e)
+		{
+			e.printStackTrace();
+		}
+		catch (Exception e)
+		{
+			e.printStackTrace();
+		}
+		return nonReactomeInstanceEdits;
+	}
+
+	/**
+	 * Gets the furthest ancestors of this Reaction-like event which does not have newInstanceEdit as one of its InstanceEdits.
+	 * This method will perform a recursive search from a reaction-like event up to a top-level pathway, searching for a pathway that
+	 * does not have a "new" InstanceEdit. A set of all such pathways is returned.
+	 * @param newInstanceEdit a "new" InstanceEdit.
+	 * @return The pathway objects.
+	 */
+	public Set<GKInstance> getFurthestAncestorsWithoutNewInstanceEdit(GKInstance newInstanceEdit)
+	{
+		List<GKInstance> parents;
+		try
+		{
+			this.ancestors.addAll(this.getFurthestAncestorWithoutNewInstanceEdit(this.getUnderlyingInstance(), newInstanceEdit));
+		}
+		catch (Exception e)
+		{
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		return this.ancestors;
+	}
+
+	/**
+	 * Private recursive method to search up the Pathway hierarchy.
+	 * This method gets the ancestors of a pathway/RLE that is furthest up the tree and also does not have the same InstanceEdit.
+	 * @param A pathway to examine.
+	 * @param newInstanceEdit the InstanceEdit to check for.
+	 * @return The pathway objects.
+	 */
+	private Set<GKInstance> getFurthestAncestorWithoutNewInstanceEdit(GKInstance pathway, GKInstance newInstanceEdit)
+	{
+		try
+		{
+			List<GKInstance> parents = (List<GKInstance>) pathway.getReferers(ReactomeJavaConstants.hasEvent);
+			if (parents == null || parents.isEmpty() )
+			{
+				// If we somehow get to the root level where there are no more parents,
+				// just return this pathway. Probably not going to happen, but it's
+				// probably a good idea to have this base case anyway.
+				this.ancestors.add(pathway);
+			}
```

Comment:
Is it problematic if the pathway has no parents but it does have the new instance edit?  That is, should the method return the pathway (even though it has the instance edit) or should it return an empty set?

Reply:
I suppose I could check for the InstanceEdit in this situation.

Reply:
I think the empty set makes a bit more sense.

Reply:
If a pathway has no parents AND has the new InstanceEdit... actually... I'm not sure that's a problem. It would mean that a top-level pathway has been modified. So maybe I should roll back to _this_ code...

Reply:
If a top-level pathway has been modified, wouldn't the empty set returned make sense because there would be no parents or ancestors found in the recursion without the new instance edit?

Reply:
The code in this method has changed quite a bit. You might want to have a look at it in its current state.
The code will return `this.ancestors` in all cases now. `this.ancestors` is initialised as an empty set, so if this method detects that there are no parents, it will return `this.ancestors` which _is_ an empty set.

But have a look at the new code, just to see.